### ISSUE TITLE
`docker pull` should prompt for login if pull fails and no current login

### ISSUE THREAD

**Issue by metalivedev** [docker/docker#7892]
### Problem 
`docker pull` on a private repository will fail with "Error: image XXXXXX/YYYYY not found" unless you're logged in with a user account that has access to the repo.

If the user has never run `docker push` (which prompts for a login) then there may be no credentials in `~/.dockercfg` and so the `docker pull` of the private repo fails with a misleading Error message (above).

### Proposed solutions
Either:

1. Force a docker hub ID for every `docker pull` **or**
2. if the docker pull fails and there are no credentials, prompt for credentials and try again. 

And:

If there are credentials, prompt with the Not Found error plus something like "Does account *username* have access to *reponame*?"

**Comment by jfrazelle** [comment]
it prompts for credentials now afaik, but ping me if otherwise

**Comment by metalivedev** [comment]
released version of 1.5 does not prompt for credentials if a pull fails. @jfrazelle 
current master version also does not prompt (tested master.dockerproject.com Client version: 1.5.0-dev Git commit (client): 9a2e58d)

```
root@testdd7892:~# docker pull rufus/secretsiege
Pulling repository rufus/secretsiege
FATA[0000] Error: image rufus/secretsiege:latest not found 

# Ok, let's add in some credentials
root@testdd7892:~# vi .dockercfg

root@testdd7892:~# docker pull rufus/secretsiege
Pulling repository rufus/secretsiege
5d6e1b594e13: Download complete 511136ea3c5a: Download complete 
d7ac5e4f1812: Download complete 
2f4b4d6a4a06: Download complete 
83ff768040a0: Download complete 
6c37f792ddac: Download complete 
e54ca5efa2e9: Download complete 
a300044931cb: Download complete 
c864c2a70e07: Download complete 
f34c40c7767e: Download complete 
869ff409816b: Download complete 
1eecc1e8c326: Download complete 
a69472484a58: Download complete 
d7877b63754e: Download complete 
6d154d4424c0: Download complete 
a93da672afb1: Download complete 
Status: Downloaded newer image for rufus/secretsiege:latest
```
